### DISCUSSION TITLE
How to subscribe to a feed?

### DISCUSSION THREAD

**Discussion by lokxii** [MarshalX/atproto#283] [Questions] [answered]
There are examples to subscribe firehose and process events. Is there anyway to subscribe to a feed instead? Are there any docs / example showing how to do so?

**Comment by MarshalX** [comment]
Hi, check this https://www.docs.bsky.app/docs/tutorials/viewing-feeds#feed-generators. Press on the "Python" tab to see examples

**Comment by lokxii** [comment]
I was expecting hooking a callback to a firehose client variation just like in this example: https://github.com/MarshalX/atproto/blob/main/examples/firehose/process_commits.py

**Comment by MarshalX** [comment, marked as answer]
No custom feeds doesn't create events in firehose. This is not possible to subscribe to updates. Custom feeds are independent services out of PDS. The only way is to pull posts

**Comment by lokxii** [comment]
Thx for the reply! I guess I have rethink how my project is going to work
